app/Export/GetImages.hs

```diff
@@ -15,43 +15,44 @@ import qualified Data.Text as T
 import System.Random
 import Svg.Generator
 import Export.ImageConversion
-import Happstack.Server (Request, rqCookies, cookieValue)
-import Data.List.Utils (replace)
 import Database.CourseQueries (getMeetingTime)
 import Database.Tables as Tables
 import Database.Persist.Sqlite (runSqlite)
 import Config (databasePath)
 import Data.Fixed (mod')
+import Data.Aeson (decode)
+import Data.ByteString.Char8 as BC (pack)
+import Data.ByteString.Lazy (fromStrict)
+import Data.Maybe (fromMaybe, fromJust)
+
 
 -- | If there is an active graph available, an image of that graph is created,
 -- otherwise the Computer Science graph is created as a default.
 -- Either way, the resulting graph's .svg and .png names are returned.
-getActiveGraphImage :: Request -> IO (String, String)
-getActiveGraphImage req = do
-    let cookies = M.fromList $ rqCookies req
-        graphName =
-            T.pack $
-            replace "-" " " $
-                maybe "Computer-Science" cookieValue (M.lookup "active-graph" cookies)
-    getGraphImage graphName (M.mapKeys T.pack $ M.map (T.pack . cookieValue) cookies)
+getActiveGraphImage :: String -> IO (String, String)
+getActiveGraphImage jsonLocalStorage = do
```

Comment:
We shouldn't refer to "local storage" in variable names in the Haskell code - from the server's point of view, it got the request data from somewhere, but it doesn't need to know where. Please choose more general names throughout your changes.